hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   Setting this register to zero creates a minimally-wide CS_N-high
+                   pulse of one-half sck cycle.'''
+          resval: 0
+        },
+        { bits: "15:0",
+          name: "CLKDIV"
+          desc: '''Core clock divider.  Slows down subsequent SPI transactions by a
+                   factor of (CLKDIV+1) relative to the core clock frequency.  The
+                   period of sck, T(sck), then becomes 2*(CLK_DIV+1)*T(core)'''
+          resval: 0
+        },
+      ]
+    },
+    { name: "COMMAND",
+      desc: "Command Register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "3:0",
+          name: "TX1_CNT",
+          desc: '''The number of 1-byte bursts to transmit in this command,
+                   at the single-line data width (8 sck clocks per cycle).
+                   All other data lines are high-Z during this command phase.
+                   This field is ignored when SPEED = 0.
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "12:4",
+          name: "TXN_CNT",
+          desc: '''The number of 1-byte bursts to transmit in this command,
+                   at the full-data width (Determined by the SPEED field).
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "16:13",
+          name: "DUMMY_CNT",
+          desc: '''Controls the number of sck dummy cycles (NOT bytes)
+                   inserted between end of the last TX byte and the beginning of
+                   the first RX byte.   No data is transmitted or received
+                   and all outputs are high-Z during this command phase.  This
+                   field is ignored when SPEED = 0''',
+          resval: "0x0"
+        },
+        { bits: "25:17",
+          name: "RX_CNT",
+          desc: '''In Dual or Quad mode, this is the number of 1-byte bursts
+                   to receive in this command.  This field is ignored when
+                   SPEED = 0, because the number of bytes received is always
+                   equal to the number of bytes transmitted in Standard SPI
+                   mode.'''
+          resval: "0x0"
+        },
+        { bits: "28:26",
+          name: "CSID",
+          desc: '''The Chip select ID of the device to receive the following
+                   command.''',
+          resval: "0x0",
+        },
+        { bits: "30:29",
+          name: "SPEED",
+          desc: '''The speed for the following command: "0" = Standard SPI. "1" = Dual SPI.
+                    "2"=Quad SPI,  "3": RESERVED.''',
+          resval: "0x0"
+        },
+        { bits: "31",
+          name: "GO",
+          hwqe: "true",
+          hwaccess: "hrw",
+          desc: '''Writing a one to this field finalizes the command and submits it to the SPI_HOST core.
+                   This field is self-clearing, so each write to this field only triggers one command.'''
+          resval: "0x0"
+        }
```

Comment:
Why do we need this field? Can't software assemble a command word locally and then send it in a single write to the COMMAND register? In which case, "GO" would essentially be the write enable for the register.

I can't think of another use-case except for assembling the fields in pieces, but that would be really painful because you'd have to do a read-modify-write over the TL bus for each field. Maybe I'm missing something?

If we do keep the field, I'd suggest a swaccess of something like r0w1s because reading it never really makes sense.

Reply:
I went back and forth on this.
My original thought is that somebody could be assembling this with byte instructions, and it would be annoying if partial commands went out?  So this just adds some finality to the process. I'm happy to remove it, though.   Happy to change it to r0w1s if that is allowed.

Reply:
My take on it is that SW probably won't want to do this since it's slow and (dusting off my old firmware engineer hat) feels icky. So I'd do whatever is easiest for the HW: probably just trigger a command on any write to the register?

Reply:
I'm not necessarily a fan of this either, however it is mostly harmless. Firmware has to do a write to the COMMAND register anyway.  They can add this one more bit and then it is all done in one write. (maybe four more instructions to mask the GO bit in, but still harmless).  Although the one thing this *does* do, is give some absolute clarity as to what sends the command.  That one bit, the magic is in *there*!

Maybe this isn't necessary, or it's out of line with the other IPs...  I'm not married to this.

Reply:
This comment is to be cited in #5157, and will hopefully be addressed in next PR, #5149.

Reply:
Resolving here, will take up in issue.